### ISSUE TITLE
Sniffing does not support IPv6 address

### ISSUE THREAD

**Issue by takaomag** [elastic/elasticsearch-py#1108]
**Elasticsearch version** : 7.6.0

**`elasticsearch-py` version** : 7.5.1 

**Description**:
Since [Transport._get_host_info(self. host_info)](https://github.com/elastic/elasticsearch-py/blob/35afe32a6dc34bf8cdd946a3723b59b2d0b375a4/elasticsearch/transport.py#L240) can not handle IPv6 address, sniffing fails to get a list of nodes upon startup.

**Comment by bartier** [comment]
@takaomag Hi!

Actually this method can handle IPv6 based on my tests, see below:
![image](https://user-images.githubusercontent.com/18057391/87234075-ccb94180-c3a3-11ea-9255-d08b9d39364a.png)

script:
```
from elasticsearch import Elasticsearch

es = Elasticsearch(['[2600:1f1e:987:e9ff:3de0:53a7:cb19:235b]'], sniff_on_start=True)

print('nodes:')
for node in es.nodes.info()['nodes']:
    print(f'name => {es.nodes.info()["nodes"][node]["name"]}, IP => {es.nodes.info()["nodes"][node]["ip"]}')
```
It is necessary to use brackets when using IPv6, like [2600:1f1e:987:e9ff:3de0:53a7:cb19:235b] or it is going to raise an error. Is this behavior you are referring to?

**Comment by takaomag** [comment]
Thanks!
It seems It had been fixed #71840d3

**Comment by yangshugithub** [comment]
> Thanks!
> It seems It had been fixed #71840d3

how to fixed it,
